nixos/modules/services/web-apps/hledger-web.nix

```diff
@@ -50,28 +74,60 @@ in {
         Base URL, when sharing over a network.
       '';
     };
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = [ "--forecast" ];
+      description = ''
+        Extra command line arguments to pass to hledger-web.
+      '';
+    };
+
   };
 
   config = mkIf cfg.enable {
-    systemd.services.hledger-web = {
+
+    users.extraUsers.hledger = {
```

Comment:
You should update to the more modern `users.users`. You should also set [isSystemUser = true;](https://github.com/NixOS/nixpkgs/pull/115332#issue-586258503) in here.

Reply:
Done! Added an `useDefaultShell` though, as the user should be easy to SSH into imho. At least that is my usecase for this module.